scheme/proto/src/main/java/cz/o2/proxima/scheme/proto/utils/ProtoUtils.java

```diff
@@ -15,75 +15,150 @@
+      Descriptor proto, T defaultValue) {
+    final Map<String, SchemaTypeDescriptor<?>> fields =
+        proto
+            .getFields()
+            .stream()
+            .collect(Collectors.toMap(FieldDescriptor::getName, ProtoUtils::convertField));
+
+    return SchemaDescriptors.structures(
+        proto.getName(),
+        fields,
+        new ProtoMessageValueAccessor<>(fields, () -> proto, () -> defaultValue));
   }
 
+  /**
+   * Convert field of proto message to type descriptor
+   *
+   * @param proto field proto descriptor
+   * @param <T> field type
+   * @return schema type descriptor
+   */
   @SuppressWarnings("unchecked")
   protected static <T> SchemaTypeDescriptor<T> convertField(FieldDescriptor proto) {
     SchemaTypeDescriptor<T> descriptor;
 
     switch (proto.getJavaType()) {
       case STRING:
-        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.strings().toTypeDescriptor();
+        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.strings();
         break;
       case BOOLEAN:
-        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.booleans().toTypeDescriptor();
+        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.booleans();
         break;
       case BYTE_STRING:
-        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.bytes().toTypeDescriptor();
+        descriptor =
+            (SchemaTypeDescriptor<T>)
+                SchemaDescriptors.bytes(
+                    new PrimitiveValueAccessor<ByteString>() {
+                      @Override
+                      public ByteString createFrom(Object object) {
+                        return ByteString.copyFrom((byte[]) object);
+                      }
+
+                      @Override
+                      public Object valueOf(ByteString value) {
+                        return value.toByteArray();
+                      }
+                    });
         break;
       case FLOAT:
-        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.floats().toTypeDescriptor();
+        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.floats();
         break;
       case DOUBLE:
-        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.doubles().toTypeDescriptor();
+        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.doubles();
         break;
       case LONG:
-        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.longs().toTypeDescriptor();
+        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.longs();
         break;
       case INT:
-        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.integers().toTypeDescriptor();
+        descriptor = (SchemaTypeDescriptor<T>) SchemaDescriptors.integers();
         break;
       case ENUM:
         descriptor =
             (SchemaTypeDescriptor<T>)
                 SchemaDescriptors.enums(
+                    new ArrayList<>(
```

Comment:
why do we needs this "extra" wrap?

Reply:
Nope, missed during refactoring. Thanks 👍